script/stop-update.lua

```diff
@@ -377,7 +421,14 @@ end
 function setLamp(trainStop, color, count)
   -- skip invalid stops and colors
   if trainStop and trainStop.lampControl.valid and ColorLookup[color] then
-    trainStop.lampControl.get_control_behavior().parameters = {parameters={{index = 1, signal = {type="virtual",name=ColorLookup[color]}, count = count }}}
+    local color_name = ColorLookup[color]
+    local stopName = trainStop.entity.backer_name
+    for _, stopID2 in pairs(global.TrainStopNames[stopName]) do

```

Comment:
setLamp should only set for one specific stop.
create setLamps or iterate in calling function

Reply:
Small snag with the lamps. I've removed the loop in setLamp but that exposed other issues.

 In activeDeliveries you keep track of the trainIDs headed towards a station. But that includes trains already at that station. Normally no problem. But now I have e.g. 4 stations and 2 trains start moving. Now all 4 station should become yellow, right? When a train arrives the station turns blue. That works too. But 2 stations remain yellow because they can't know the train is already at another stop.

I think I have to change the logic to only count trains on the way to a station but not yet arrived and keep this in a global table indexed by station name, lets call that pendingTrains. When the first train is added to pendingTrains all stations with the same name turn yellow. Stations with a train present turn blue and the train is removed from pendingTrains. When the last train is removed from pendingTrains all stations of that name with no train turn green. When a train leaves the station turns yellow or green depending on pendingTrains.

How does that sound?

Reply:
The logic behind setting colors and having the right count is quite complex. Not only does it store how many trains are en route it also includes error state handling.
If you touch that, I'll have to re-read not only your changes but also my own logic I wrote over a year ago and do extensive testing to see if everything still works as intended. 
Currently I'm way too engaged with Satisfactory and don't fancy digging this deep into LTN.

For your question:
Blue always means a train is parked at this very stop. Same name stops should keep showing yellow.
All stops of the same name must have the same train count at all times.

Reply:
Stops should show yellow even if no train is incoming because it already arrived at a station with the same name?

Reply:
Blue means a train is parked at this stop.
Basically blue signal is there to distinguish between LTN trains parked at a stop and normal trains.